daft/filesystem.py

```diff
@@ -57,35 +52,27 @@ class ListingInfo:
     rows: int | None = None
 
 
-def _get_s3fs_kwargs() -> dict[str, Any]:
-    """Get keyword arguments to forward to s3fs during construction"""
-
-    try:
-        import botocore.session
-    except ImportError:
-        logger.error(
-            "Error when importing botocore. Install getdaft[aws] for the required 3rd party dependencies to interact with AWS S3 (https://www.getdaft.io/projects/docs/en/latest/learn/install.html)"
-        )
-        raise
-
-    kwargs = {}
-
-    # If accessing S3 without credentials, use anonymous access: https://github.com/Eventual-Inc/Daft/issues/503
-    credentials_available = botocore.session.get_session().get_credentials() is not None
-    if not credentials_available:
-        logger.warning(
-            "AWS credentials not found - using anonymous access to S3 which will fail if the bucket you are accessing is not a public bucket. See boto3 documentation for more details on configuring your AWS credentials: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html"
-        )
-        kwargs["anon"] = True
-
-    # kwargs["default_fill_cache"] = False
+def get_filesystem(protocol: str, **kwargs) -> fsspec.AbstractFileSystem:
+    if protocol == "s3" or protocol == "s3a":
+        try:
+            import botocore.session
```

Comment:
does this mean that we need botocore even when we are doing reads with the native downloader?

Reply:
It shouldn't - `get_filesystem` is called right now only in tests and an our non-native URL downloads (which uses fsspec)